pubsub-mapped-api/src/test/java/com/google/pubsub/clients/consumer/KafkaConsumerTest.java

```diff
@@ -0,0 +1,333 @@
+package com.google.pubsub.clients.consumer;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Empty;
+import com.google.protobuf.Timestamp;
+import com.google.pubsub.kafkastubs.common.KafkaException;
+import com.google.pubsub.kafkastubs.common.serialization.IntegerSerializer;
+import com.google.pubsub.kafkastubs.common.serialization.StringSerializer;
+import com.google.pubsub.kafkastubs.consumer.ConsumerRecord;
+import com.google.pubsub.kafkastubs.consumer.ConsumerRecords;
+import com.google.pubsub.v1.AcknowledgeRequest;
+import com.google.pubsub.v1.DeleteSubscriptionRequest;
+import com.google.pubsub.v1.ListTopicsRequest;
+import com.google.pubsub.v1.ListTopicsResponse;
+import com.google.pubsub.v1.PublisherGrpc.PublisherImplBase;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.PullRequest;
+import com.google.pubsub.v1.PullResponse;
+import com.google.pubsub.v1.ReceivedMessage;
+import com.google.pubsub.v1.SubscriberGrpc;
+import com.google.pubsub.v1.SubscriberGrpc.SubscriberFutureStub;
+import com.google.pubsub.v1.SubscriberGrpc.SubscriberImplBase;
+import com.google.pubsub.v1.Subscription;
+import com.google.pubsub.v1.Topic;
+import io.grpc.stub.StreamObserver;
+import io.grpc.testing.GrpcServerRule;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.regex.Pattern;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.powermock.core.classloader.annotations.SuppressStaticInitializationFor;
+
+
+@RunWith(JUnit4.class)
+@SuppressStaticInitializationFor("com.google.pubsub.common.ChannelUtil")
+public class KafkaConsumerTest {
+
+  @Rule
+  public final GrpcServerRule grpcServerRule = new GrpcServerRule().directExecutor();
+
+  private KafkaConsumer<Integer, String> consumer;
+
+  @Before
+  public void setUp() throws ExecutionException, InterruptedException {
+    ConsumerConfig config = getConsumerConfig();
+    consumer = new KafkaConsumer<>(config, null,
+        null, grpcServerRule.getChannel(), null);
+  }
+
+  @Test
+  public void testSubscribe() {
```

Comment:
Junit 4 tests don't use the "test" prefix in the method name.  This is a JUnit3-ism.